CMakeLists.txt

```diff
@@ -204,7 +204,7 @@ include_directories(SYSTEM
   ${OPENGL_INCLUDE_DIR}
   ${PYTHON_INCLUDE_PATH}
 )
-include_directories(src ${catkin_INCLUDE_DIRS})
+include_directories(src SYSTEM ${catkin_INCLUDE_DIRS})
```

Comment:
I see, I did not take into account multiple workspaces. That issue is unfortunate, as it doesn't seem like something that can easily be worked around.

My main hesitation with using pragmas is that they should be maintained properly to stay in sync when new warnings are added by the external dependency. Also, removing unneeded pragmas becomes difficult since it would mean reintroducing warnings when compiling with older versions of the external dependency. On the other hand `-isystem` more clearly communicates the intent ("these headers are not mine, don't blame me") and doesn't require maintenance.

Is there perhaps a pragma for gcc and clang which has the same effect on an included header as `-isystem`? That would fix the problem without adding a maintenance burden while avoiding include path order issues.